### PR TITLE
Add year service for Ruby

### PR DESCRIPTION
**Pull request by ajvondrak**
It looks like every other language had this service. Guess Ruby just
missed the boat. Oops!

There's already a smattering of framework choices represented in the
other Ruby services:

- frontend uses Rails and OTel
- message-service uses Rails and the Beeline
- name-service uses Sinatra and the Beeline

I decided to expand the list with year-service by using the Grape
framework. By itself, it's not much more interesting than a plain Rack
or Sinatra app. But it does have builtin ActiveSupport notifications
that we can readily instrument with the Beeline.

The other year services don't seem to generate manual spans, so I just
kept with the barebones Rack auto-instrumentation (plus Grape's
notifications).

An example trace

**Full disclosure:** I haven't gotten Tilt up and running, so I don't know if/how it breaks. The Dockerfile seems to run fine on my localhost, but I'd also be skeptical of it as a rule. 😂

### CODE REVIEW

**Comment by robbkidd** [comment]
tenor-134289674

**Comment by robbkidd** [comment]
Mmm. Purdy.

a trace of a web request flowing through 4 different Ruby microservices

**Comment by robbkidd** [comment]
> The other year services don't seem to generate manual spans ...

I think it would be good to update that! [The Java year implementation does have a manual span](https://github.com/honeycombio/example-greeting-service/blob/main/java/year-service/src/main/java/io/honeycomb/examples/javaotlp/YearService.java#L22), but most of the others don't. It is a silly wrapping of "however long it takes to pick a year."

So, possibly:

```ruby
  get "/year" do
    year = nil
    Honeycomb.start_span(name: "🗓 get-a-year ✨") do |_span|
      sleep rand(0..0.005)
      year = (2015..2020).to_a.sample
    end
    year
  end
```

File: ruby/year-service/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM ruby:3.0
```
Comment (robbkidd) [inline ruby/year-service/Dockerfile:1]: Ooo. We fancy. ✨

**Review by robbkidd — APPROVED** [review]
👍🏻  💖    I love another framework example!

gobble down the grape

Works pretty much as is on my kubify (kustomize) branch. I suggest that we can figure out the details of what to update Tilt (or whatever runner we settle upon) in another PR.

**Comment by ajvondrak** [comment]
> > The other year services don't seem to generate manual spans ...
> 
> I think it would be good to update that!

Done:
Example trace with the new get-a-year span

**Review by robbkidd — APPROVED** [review]
tenor-235307401

**Pull request merged**